src/uthash.h

```diff
@@ -983,14 +1058,26 @@ do {
           }                                                                      \
           if ((dst) == NULL) {                                                   \
             DECLTYPE_ASSIGN(dst, _elt);                                          \
-            HASH_MAKE_TABLE(hh_dst, dst);                                        \
+            HASH_MAKE_TABLE(hh_dst, dst, _mem_ok);                               \
+            if (HASH_NOMEM_OK && !_mem_ok) {                                     \
+              uthash_fatal(_elt);                                                \
+              dst = NULL;                                                        \
+              continue;                                                          \
+            }                                                                    \
```

Comment:
One thing you can/should do here to simplify the code is to introduce a macro at the top of uthash.h like this:

    #define IF_HASH_NOMEM_OK(x) x
    #define IF_HASH_NOMEM_OK(x)

and then down here,

    HASH_MAKE_TABLE(hh_dst, dst, _mem_ok);
    IF_HASH_NOMEM_OK(
      if (!_mem_ok) {
        uthash_fatal(_elt);
        dst = NULL;
        continue;
      }
    )